sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -6,30 +6,33 @@ import nise from "nise";
 import {
   isBrowser,
   blobToString,
-  escapeRegExp,
   env,
   TestInfo,
   parseUrl,
   isPlaybackMode,
   isRecordMode,
-  findRecordingsFolderPath
+  findRecordingsFolderPath,
+  applyReplacementDictionary,
+  applyReplacementFunctions,
+  ReplacementFunctions,
+  ReplacementDictionary
 } from "./utils";
 import { customConsoleLog } from "./customConsoleLog";
 
 let nock: any;
 
-let replaceableVariables: { [x: string]: string } = {};
-export function setReplaceableVariables(a: { [x: string]: string }): void {
-  replaceableVariables = a;
+let replaceableVariables: ReplacementDictionary = {};
+export function setReplaceableVariables(replacements: ReplacementDictionary): void {
+  replaceableVariables = replacements;
   if (isPlaybackMode()) {
     // Providing dummy values to avoid the error
-    Object.keys(a).map((k) => {
-      env[k] = a[k];
+    Object.keys(replacements).map((k) => {
```

Comment:
You're not mapping, so this should be forEach. But even better is an actual loop - `for (const k of Object.keys(replacements)) { }`. You can delete the Object.keys part if you go with the map approach, too.